consensus/consensus-types/src/proposal_msg.rs

```diff
@@ -27,17 +29,8 @@ impl<T: Payload> TryFrom<network::proto::Proposal> for ProposalUncheckedSignatur
     type Error = failure::Error;
 
     fn try_from(proto: network::proto::Proposal) -> failure::Result<Self> {
-        let proposal = proto
-            .proposed_block
-            .ok_or_else(|| format_err!("Missing proposed_block"))?
-            .try_into()?;
-        let sync_info = proto
-            .sync_info
-            .ok_or_else(|| format_err!("Missing sync_info"))?
-            .try_into()?;
-        Ok(ProposalUncheckedSignatures(ProposalMsg::new(
-            proposal, sync_info,
-        )))
+        let proposal: ProposalMsg<T> = lcs::from_bytes(&proto.bytes)?;
```

Comment:
you probably don't need this line?
`Ok(ProposalUncheckedSignatures(lcs::from_bytes(&proto.bytes)?)`

Reply:
oh, hmm... for some reason I thought rust would think it is a ProposalUncheckedSignatures and not a Proposal, good call.